server/src/main/java/org/candlepin/dto/shim/ProductDataTranslator.java

```diff
@@ -70,42 +79,51 @@ public ProductDTO populate(ModelTranslator modelTranslator, ProductData source,
-        dest.setAttributes(source.getAttributes());
-        dest.setDependentProductIds(source.getDependentProductIds());
-        dest.setHref(source.getHref());
-        dest.setLocked(source.isLocked());
-
+        dest.id(source.getId())
+            .uuid(source.getUuid())
+            .name(source.getName())
+            .multiplier(source.getMultiplier())
+            .created(Util.toDateTime(source.getCreated()))
+            .updated(Util.toDateTime(source.getUpdated()))
+            .attributes(toAttributes(source.getAttributes()))
+            .productContent(new HashSet<>())
+            .branding(new HashSet<>())
+            .dependentProductIds(toSet(source))
+            .href(source.getHref())
+            .locked(source.isLocked());
 
         if (modelTranslator != null) {
             Collection<ProductContentData> productContentData = source.getProductContent();
-            dest.setProductContent(null);
-            if (productContentData != null) {
+            if (productContentData != null && !productContentData.isEmpty()) {
                 ObjectTranslator<ContentData, ContentDTO> contentTranslator = modelTranslator
                     .findTranslatorByClass(ContentData.class, ContentDTO.class);
-
-                for (ProductContentData pcd : productContentData) {
-                    if (pcd != null && pcd.getContent() != null) {
-                        ContentDTO dto = contentTranslator.translate(modelTranslator, pcd.getContent());
-                        dest.addContent(dto, pcd.isEnabled());
+                Set<ProductContentDTO> dtos = new HashSet<>();
+                for (ProductContentData productContent : productContentData) {
+                    if (productContent != null && productContent.getContent() != null) {
+                        ContentDTO dto = contentTranslator
+                            .translate(modelTranslator, productContent.getContent());
+                        dtos.add(createContent(dto, productContent.isEnabled()));
                     }
                 }
+                dest.productContent(dtos);
+            }
+            else {
+                dest.productContent(Collections.emptySet());
             }
 
             Collection<Branding> productBrandings = source.getBranding();
-            dest.setBranding(null);
-            if (productBrandings != null) {
+            if (productBrandings != null && !productBrandings.isEmpty()) {
+                Set<BrandingDTO> dtos = new HashSet<>();
                 for (Branding brand : productBrandings) {
                     if (brand != null) {
-                        dest.addBranding(modelTranslator.translate(brand, BrandingDTO.class));
+                        BrandingDTO dto = modelTranslator.translate(brand, BrandingDTO.class);
+                        if (isNullOrIncomplete(dto)) {
+                            throw new IllegalArgumentException("branding is null or incomplete");
```

Comment:
Same here: no need for validation at this point

Reply:
Done.